core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -220,7 +232,10 @@ private[chisel3] trait HasId extends InstanceId {
     case Some(p) => p._component match {
       case Some(c) => _ref match {
         case Some(arg) => arg fullName c
-        case None => computeName(None, None).get
+        case None => computeName(None, None) match {
+          case Some(name) => name
+          case None => throwException("trying to access a name that cannot be computed")
+        }
```

Comment:
@jackkoenig this is my attempt at throwing an exception for annotating stuff without names. Please let me know if there's a better place to put this/word this.

Reply:
Seems reasonable. We probably ought to just error instead of calling `computeName(None, None)` but that's unrelated to this PR so I think what you're here doing is good.

Reply:
resolving b/c i switched to naming everything